buddy_alloc.h

```diff
@@ -1081,7 +1124,7 @@ static struct buddy_tree *buddy_tree_init(unsigned char *at, uint8_t order) {
     struct buddy_tree *t = (struct buddy_tree*) at;
     memset(at, 0, size);
     t->order = order;
-    t->upper_pos_bound = 1u << t->order;
+    t->upper_pos_bound = ((size_t)1u) << t->order;
```

Comment:
I went ahead and wrapped all those shifts with cast to size_t

Reply:
This is making it somewhat cumbersome to read. What's the behavior of MSVC without the cast ?

Reply:
its about size_t being 32-bit or 64-bit, I'm casting it to appropriate type to avoid overflow

Reply:
But overflow cannot really happen here - the order of the tree is way smaller. Although compilers don't understand that currently - that's why I had to make `1 -> 1u`, so that they clang won't complain about UB on signed overflow.

Reply:
Hm, I still want to have 0-warnings, I would try using __assume / __builtin_assume to hint the compiler

Reply:
Marking with assume didn't help, I've reverted that change